log_outgoing_requests/utils.py

```diff
@@ -0,0 +1,155 @@
+# Handler utilities
+#
+def _get_content_length(http_obj: Union[Request, Response]) -> Optional[int]:
+    """
+    Try to determine the size of a request/response content.
+
+    Try `Content-Length` header first. If not present, try to
+    determine the size by reading `len(body)` or `len(content)`.
+    The entire content is thereby read into memory (the assumption
+    being that the content will eventually be consumed anyway).
+    """
+    content_length = http_obj.headers.get("Content-Length", "")
+
+    try:
+        content_length = int(content_length)
+    except ValueError:
+        if isinstance(http_obj, Response):
+            try:
+                content_length = len(http_obj.content)
+            except TypeError:
+                return None
+        else:
+            try:
+                content_length = len(http_obj.body)
+            except TypeError:
+                return None
+
+    return content_length
+
+
+def _check_content_length(
+    http_obj: Union[Request, Response],
+    content_length: Optional[int],
+    config: "OutgoingRequestsLogConfig",
+) -> bool:
+    """
+    Check `content_length` against settings.
+
+    If `content_length` could not be determined, the test passes with a
+    warning.
+    """
+
+    if content_length is None:
+        logger.warning(
+            "Content length of the request/response (netloc: %s) could not be determined."
+            % http_obj.netloc
+        )
+        return True
+
+    max_content_length = config.max_content_length
+
+    return content_length <= max_content_length
+
+
+def _check_content_type(
+    content_type: Optional[str], config: "OutgoingRequestsLogConfig"
+) -> bool:
+    """
+    Check `content_type` against settings.
+
+    The content types specified under `LOG_OUTGOING_REQUESTS_CONTENT_TYPES`
+    are split into two groups. For regular types (without wildcard), check
+    if `content_type` is included in the list. For types specified with a
+    wildcard ('*'), check if `content_type` is a substring of any string
+    contained in the list.
+    """
+    if content_type is None:
+        return False
+
+    allowed_mime_types: Iterable[str] = settings.LOG_OUTGOING_REQUESTS_CONTENT_TYPES
+    regulars = [item for item in allowed_mime_types if not item.endswith("*")]
+    wildcards = [item for item in allowed_mime_types if item.endswith("*")]
+
+    if content_type in regulars:
+        return True
+
+    return any(content_type.startswith(pattern[:-1]) for pattern in wildcards)
+
+
+def is_content_admissible_for_saving(
+    http_obj: Union[Request, Response],
+    config: "OutgoingRequestsLogConfig",
+) -> bool:
+    """Determine if content should be saved"""
+
+    content_type = http_obj.headers.get("Content-Type")
+    content_length = _get_content_length(http_obj)
+
+    return _check_content_type(content_type, config) and _check_content_length(
+        http_obj, content_length, config
+    )
+
+
+def get_encoding(request: Request) -> str:
```

Comment:
encoding is for both request and response, but I would make this function operate on the value of `Content-Type` as per my earlier remarks/examples.